File: app/src/main/java/com/confessionsearch/release1/data/notes/NoteRepository.java
package com.confessionsearch.release1.data.notes;

import android.content.Context;

import androidx.lifecycle.LiveData;

import com.confessionsearch.release1.databaseshelpers.notesDBClassHelper;

import java.util.List;

public class NoteRepository {

    private final notesDBClassHelper notesDB;

    public NoteRepository(Context context)
    {
        notesDB= notesDBClassHelper.getInstance(context);
    }
    public void insertNote(Notes note){
new InsertAsync(notesDB.getNoteDao()).execute(note);
    }
    public void updateNote(Notes note){
        new UpdateAsync(notesDB.getNoteDao()).execute(note);
    }

    public void deleteNote(Notes note){
        new DeleteAsync(notesDB.getNoteDao()).execute(note);
    }


    public LiveData<List<Notes>>fetchNotes(){
        return notesDB.getNoteDao().fetchNotes();
    }
}

File: app/src/main/java/com/confessionsearch/release1/data/notes/NoteRepository.kt
package com.confessionsearch.release1.data.notes

import android.content.Context
import androidx.lifecycle.LiveData
import com.confessionsearch.release1.databaseshelpers.notesDBClassHelper
import com.confessionsearch.release1.databaseshelpers.notesDBClassHelper.Companion.getInstance

class NoteRepository(context: Context?) {
    private val notesDB: notesDBClassHelper? = getInstance(context!!)
    fun insertNote(note: Notes?) {
        InsertAsync(notesDB!!.noteDao!!).execute(note)
    }

    fun updateNote(note: Notes?) {
        UpdateAsync(notesDB!!.noteDao!!).execute(note)
    }

    fun deleteNote(note: Notes?) {
        DeleteAsync(notesDB!!.noteDao!!).execute(note)
    }

    fun fetchNotes(): LiveData<List<Notes>> {
        return notesDB!!.noteDao!!.fetchNotes()
    }

}